server.go

```diff
@@ -70,13 +71,22 @@ func (s *Server) CheckHealth(traceOn bool, logger Logger) {
 	var secondsBehindMaster, openConnections, runningConnections *int
 
 	// prevent concurrently checks on same server (slow queries/network)
-	if s.isChecking {
+	s.mtx.RLock()
```

Comment:
```suggestion
	s.RLock()
```

E o mesmo nos demais